packages/hoppscotch-backend/src/infra-config/infra-config.service.ts

```diff
@@ -182,6 +184,51 @@ export class InfraConfigService implements OnModuleInit {
     }
   }
 
+  /**
+   * Check if the service is configured or not
+   * @param service Service can be Auth Provider, Mailer, Audit Log etc.
+   * @returns Either true or false
+   */
+  async isServiceConfigured(service: AuthProvider) {
+    switch (service) {
+      case AuthProvider.GOOGLE:
+        if (
```

Comment:
can further branching be written in a more abstract way ?

Reply:
Can you give any suggestions?

Reply:
A helper method that could abstract away the common logic, where we could pass values as arguments maybe ?

Reply:
How this looks, @ankitsridhar16?
```ts
isServiceConfigured(service: AuthProvider | 'AUDIT_LOG') {
  const configService = this.configService;

  switch (service) {
    case AuthProvider.GOOGLE:
      return configService.get<string>('INFRA.GOOGLE_CLIENT_ID') &&
             configService.get<string>('INFRA.GOOGLE_CLIENT_SECRET');

    case AuthProvider.GITHUB:
      return configService.get<string>('INFRA.GITHUB_CLIENT_ID') &&
             configService.get<string>('INFRA.GITHUB_CLIENT_SECRET');

    case AuthProvider.MICROSOFT:
      return configService.get<string>('INFRA.MICROSOFT_CLIENT_ID') &&
             configService.get<string>('INFRA.MICROSOFT_CLIENT_SECRET');

    case AuthProvider.EMAIL:
      return configService.get<string>('INFRA.MAILER_SMTP_URL') &&
             configService.get<string>('INFRA.MAILER_ADDRESS_FROM');

    case AuthProvider.SAML:
      return configService.get<string>('INFRA.SAML_ISSUER') &&
             configService.get<string>('INFRA.SAML_CERT') &&
             configService.get<string>('INFRA.SAML_ENTRY_POINT');

    case 'AUDIT_LOG':
      return !!configService.get<string>('INFRA.CLICKHOUSE_HOST');

    default:
      return false;
  }
}
```

Reply:
👍